neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/impl/BaseNeo4jSchemaDefiner.java

```diff
@@ -181,10 +192,42 @@ public void initializeSchema(SchemaDefinitionContext context) {
 		List<Sequence> sequences = sequences( context.getDatabase() );
 
 		createSequences( sequences, context.getAllIdSourceKeyMetadata(), provider );
+		createIndexesIfMissing( provider, indexSpecs );
 		createEntityConstraints( provider, context.getDatabase(), sessionFactoryImplementor.getProperties() );
 	}
 
+	private void validateIndexSpecs(SchemaDefinitionContext context) {
+		Database database = context.getDatabase();
+		for ( Namespace namespace : database.getNamespaces() ) {
+			for ( Table table : namespace.getTables() ) {
+				if ( table.isPhysicalTable() ) {
+					Label label = label( table.getName() );
+					Iterator<Index> indexIterator = table.getIndexIterator();
+					while ( indexIterator.hasNext() ) {
+						addIndex( label, indexIterator.next() );
+					}
+				}
+			}
+		}
+	}
+
+	private void addIndex(Label label, Index index) {
+		if ( index != null ) {
+			if ( index.getName() != null ) {
+				log.cannotSetNameForIndex( index.getName() );
```

Comment:
Why do we have both cannotSetNameForIndex and warnf?

Reply:
Definitely an error while rebasing. There should only be the cannotSetNameForIndex.